app/src/main/java/com/example/filemanager/AllFilesViewModel.kt

```diff
@@ -1,21 +1,32 @@
 package com.example.filemanager
 
 import android.os.Environment
-import android.util.Log
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import java.io.File
 
 class AllFilesViewModel() : ViewModel() {
 
     val files = MutableLiveData<List<ListModel>>()
+    private val rootDirectory: ListModel = ListModel(Environment.getExternalStorageDirectory())
+    var currentDirectory: ListModel = rootDirectory
 
-    fun fetchFiles() {
-        val path = Environment.getExternalStorageDirectory().absolutePath ?: return
-        Log.d("PATH", path)
-        val files2 = File(path).listFiles()?.map {
+    fun listRootDirectoryFiles() {
+        files.postValue(listFiles(rootDirectory))
+    }
+
+    fun listFiles(currentItem: ListModel): List<ListModel> {
+        return currentItem.file.listFiles()?.map {
```

Comment:
can also create a method for list files in ListModel class that will return a list of `ListModel's`and here you can simply call: `return currentItem.listFiles()`